src/session/content/message_row/media_picture.rs

```diff
@@ -0,0 +1,189 @@
+use gtk::{gdk, glib, graphene, prelude::*, subclass::prelude::*};
+
+const MIN_WIDTH: i32 = 100;
+const MIN_HEIGHT: i32 = 100;
+const MAX_HEIGHT: i32 = 400;
+
+mod imp {
+    use super::*;
+    use once_cell::sync::Lazy;
+    use std::cell::{Cell, RefCell};
+
+    #[derive(Debug, Default)]
+    pub struct MediaPicture {
+        pub paintable: RefCell<Option<gdk::Paintable>>,
+        pub aspect_ratio: Cell<f64>,
+    }
+
+    #[glib::object_subclass]
+    impl ObjectSubclass for MediaPicture {
+        const NAME: &'static str = "ContentMediaPicture";
+        type Type = super::MediaPicture;
+        type ParentType = gtk::Widget;
+    }
+
+    impl ObjectImpl for MediaPicture {
+        fn properties() -> &'static [glib::ParamSpec] {
+            static PROPERTIES: Lazy<Vec<glib::ParamSpec>> = Lazy::new(|| {
+                vec![
+                    glib::ParamSpec::new_object(
+                        "paintable",
+                        "Paintable",
+                        "The paintable of the media",
```

Comment:
I think that alone `media` has no article. But not sure as I'm not a native speaker.
I would rather call it "the media object" or "the media type".
Same applies to the next parameter.

Reply:
Well, `media` is a noun and it is also used for indicating e.g. media files, so I think it's correct (also, it's not an object or a type).

Reply:
I still think you cannot use an article in this case. Else, you would refer to media as a whole (all newspapers, the internet, etc.). Why it's not an object? Everything in the world is an object. :-P
But let's not get into nitpicking. :-)